tex/gregoriotex-syllable.tex

```diff
@@ -919,20 +927,53 @@
     \kern\dimexpr(\gre@dimen@enddifference% move from end of text to end of bar line
       -\wd\gre@box@syllablenotes)\relax % back up from end of bar line to beginning
     \GreNoBreak%
+    \global\let\gre@newlinecommonsaved\gre@newlinecommon %
+    \global\let\gre@newlinecommon\gre@newlinecommondelayed %
+    \xdef\gre@newlinearg{-1}%
     #9%
+    \global\let\gre@newlinecommon\gre@newlinecommonsaved %
     \GreNoBreak%
-    %move to the beginning of the notes of the next syllable
-    \gre@hskip\glueexpr(-\wd\gre@box@syllablenotes/2% back up to middle of notes
-      -\gre@dimen@bar@shift% go back from actual middle to nominal middle of bar line
-      +\gre@skip@bar@allocation/2)\relax % go from nominal middle to the start of the next notes
-    \ifdim\gre@skip@nextbegindifference < 0pt\relax%
-      %we need to move back to where the text for the next syllable should start
-      \GreNoBreak %
-      \kern\gre@skip@nextbegindifference\relax%
+    % get into position to place the penalty
+    \ifdim\gre@dimen@enddifference < 0pt\relax%
+      % the text extends past the notes, so we need to get to the end of the text
+      \kern-\gre@dimen@enddifference%
+      \GreNoBreak%
     \fi%
-    %finally we end by adding the penalty
-    \gre@debugmsg{barspacing}{after bar penalty: \greendafterbarpenalty}%
-    \gre@penalty{\greendafterbarpenalty}%
+    \ifgre@eolshiftsenabled%
+      %this shift will shorten the line somewhat to allow the lyrics to extend under the custos should TeX decide to insert a line break after the syllable being ended
+      \gre@calculate@eolshift{\dimexpr(\gre@dimen@enddifference-\gre@skip@bar@lastskip)\relax }%
```

Comment:
I think we can do without it... it will just make things more complex... If we go in that direction, the best would be that the C code is writing the type of bar, otherwise things will be too complex for no benefit